santoku/aws/test_sqs_handler.py

```diff
@@ -1,83 +1,100 @@
-            aws_access_key_id="testing",
-            aws_secret_access_key="testing",
-            aws_session_token="testing",
-            region_name=REGION,
-        )
-        self.client = boto3.client(
-            "sqs",
-            aws_access_key_id="testing",
-            aws_secret_access_key="testing",
-            aws_session_token="testing",
-            region_name=REGION,
-        )
-
-        # Create a standard queue.
-        try:
-            self.client.get_queue_url(QueueName=STANDARD_SQS_NAME)
-        except exceptions.ClientError:
-            # By catching the exception we check the queue certainly does not exist.
-            pass
-        else:
-            raise EnvironmentError(f"{STANDARD_SQS_NAME} should not exist.")
-        self.client.create_queue(QueueName=STANDARD_SQS_NAME)
-
-        # Create a FIFO queue.
-        try:
-            self.client.get_queue_url(QueueName=FIFO_SQS_NAME)
-        except exceptions.ClientError:
-            pass
-        else:
-            raise EnvironmentError(f"{FIFO_SQS_NAME} should not exist.")
-        self.client.create_queue(QueueName=FIFO_SQS_NAME, Attributes={"FifoQueue": "true"})
-
-    def teardown_method(self):
-        delete_queue(sqs_client=self.client, queue_name=STANDARD_SQS_NAME)
-        delete_queue(sqs_client=self.client, queue_name=FIFO_SQS_NAME)
-        self.mock_sqs.stop()
-
-    def test_check_queue_existence(self):
+    def test_check_queue_existence(self, sqs_handler, standard_queue):
         # Test an existing queue. Success expected.
-        assert self.sqs_handler.check_queue_existence(queue_name=STANDARD_SQS_NAME)
+        assert sqs_handler.get_queue_url(queue_name=standard_queue)
 
         # Test a non-existent queue. Failure expected.
-        assert not self.sqs_handler.check_queue_existence(queue_name="WRONG_QUEUE_NAME")
+        with pytest.raises(exceptions.ClientError, match="The specified queue does not exist"):
+            assert sqs_handler.get_queue_url(queue_name="WRONG_QUEUE_NAME")
 
-    def test_check_queue_is_fifo(self):
+    def test_check_queue_is_fifo(self, sqs_handler, standard_queue, fifo_queue):
         # Test a fifo queue. Success expected.
-        assert self.sqs_handler.check_queue_is_fifo(queue_name=FIFO_SQS_NAME)
+        assert sqs_handler.check_queue_is_fifo(queue_name=fifo_queue)
 
         # Test a standard queue. Failure expected.
-        assert not self.sqs_handler.check_queue_is_fifo(queue_name=STANDARD_SQS_NAME)
+        assert not sqs_handler.check_queue_is_fifo(queue_name=standard_queue)
 
-    def test_get_queue_url(self):
+    def test_get_queue_url(self, sqs_handler, standard_queue):
         # Test getting the name of a queue that does exist. Success expected.
-        obtained_url = self.sqs_handler.get_queue_url(queue_name=STANDARD_SQS_NAME)
+        obtained_url = sqs_handler.get_queue_url(queue_name=standard_queue)
+        region = os.environ["AWS_DEFAULT_REGION"]
```

Comment:
should we check and throw `MissingEnvironmentVariables`? I can't see it anywhere else but I might have missed it.

Reply:
It's set on `aws_credentials` fixture, which is a dependency of the `sqs_handler` fixture, that is a dependency of this test.